src/core/Form/Radiobutton/Radiobutton.baseStyles.tsx

```diff
@@ -0,0 +1,159 @@
+import { css } from 'styled-components';
+import { withSuomifiTheme, TokensAndTheme } from '../../theme';
+import { disabledCursor } from '../../../components/utils/css';
+import { element, font } from '../../theme/reset';
+import { focus } from '../../theme/utils/focus';
+
+/* stylelint-disable no-descending-specificity */
+export const baseStyles = withSuomifiTheme(
+  ({ theme }: TokensAndTheme) => css`
+    ${element({ theme })}
+    ${font({ theme })('bodyText')}
+
+  &.fi-radiobutton {
+    & .fi-radiobutton_hintText {
+        display: block;
+        padding-left: 26px;
+        color: ${theme.colors.depthDark1};
+        ${theme.typography.bodyTextSmall};
+    }
+
+    & .fi-radiobutton_input {
+      opacity: 0;
+      position: absolute;
+      margin: 8px 0 0 3px;
+
+      + label {
+        position: relative;
+        display: inline-block;
+        cursor: pointer;
+        min-height: 27px;
+        line-height: 1.5em;
+        padding-left: 26px;
+
+        &::before {
+          content: '';
+          position: absolute;
+          display: inline-block;
+          left: 0;
+          top: 5px;
+          width: 16px;
+          height: 16px;
+          border-radius: 50%;
+          border: 1px solid ${theme.colors.depthBase};
+          background: transparent;
+        }
+
+        &::after {
+          content: '';
+          position: absolute;
+          display: inline-block;
+          left: 5px;
+          top: 10px;
+          border-radius: 50%;
+          width: 8px;
+          height: 8px;
+        }
+      }
+
+      &:checked {
+        + label:before {
+          border: 1px solid ${theme.colors.highlightBase};
+        }
+
+        + label:after {
+          background: ${theme.colors.highlightBase};
+        }
+      }
+
+      &:focus {
+        + label::before {
+          ${focus({ theme, noPseudo: true, variant: 'boxShadow' })}
+          border-radius: 50%;
+        }
+      }
+
+      &:disabled {
+        + label {
+          ${disabledCursor}
+          color: ${theme.colors.depthBase};
+        }
+
+        + label:before {
+          border: 1px solid ${theme.colors.depthLight1};
+        }
+
+        + label:after {
+          background: ${theme.colors.depthLight3};
+          width: 16px;
+          height: 16px;
+          top: 6px;
+          left: 1px;
+        }
+      }
+
+      &:disabled:checked {
+        + label:after {
+          background: ${theme.colors.depthBase};
+          width: 8px;
+          height: 8px;
+          top: 10px;
+          left: 5px;
+        }
```

Comment:
Both variants, `small` and `large` uses this kind of styling. When radiobutton is disabled **and** selected, it is slightly different than in designs. I could not find a way to make it look exactly same and not worth the effort.
But if this is a real issue, this needs more investigation.

Reply:
Maybe check with PAMU team as well?

Reply:
Asked from the PAMU and to keep it consistent with e.g TextInput, they slight grey color should be with disabled and checked as in the design (background is now white when it is selected and the component is disabled)

Reply:
So this should now be exactly as specified in designs?

Reply:
No, because currently it is white/transparent. Haven't yet made change to this part.

Reply:
As discussed yesterday, this should be achieved by simply adding `background-color: ${theme.colors.depthLight3};` to the disabled `:before` style.

Reply:
This is now changed, just with a little twist: using `background` to set the color to keep it consistent.